src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -113,5 +149,188 @@ private void InitializeScope(IStep step)
             var scopeName = step.ExecutionContext.ScopeName;
             step.ExecutionContext.ExpressionValues["steps"] = stepsContext.GetScope(scopeName);
         }
+
+        private async Task RunStepsAsync(List<IStep> compositeSteps)
```

Comment:
To track the overall step-result, you need to do something like stepsrunner does to track the overall job-result:

Look at this code in StepsRunner since it's a similar problem:

```
                // Update the job result.
                if (step.ExecutionContext.Result == TaskResult.Failed)
                {
                    Trace.Info($"Update job result with current step result '{step.ExecutionContext.Result}'.");
                    jobContext.Result = TaskResultUtil.MergeTaskResults(jobContext.Result, step.ExecutionContext.Result.Value);
                    jobContext.JobContext.Status = jobContext.Result?.ToActionResult();
                }
                else
                {
                    Trace.Info($"No need for updating job result with current step result '{step.ExecutionContext.Result}'.");
                }
```

Reply:
(as talked in our meeting, we decided to follow a short-circuiting logic: https://github.com/actions/runner/pull/591#discussion_r456004813)